### ISSUE TITLE
[GUI] Change of behaviour in interactive controls

### ISSUE THREAD

**Issue by Camille-K** [sofa-framework/sofa#3510]
Following a recent update of my fork from SOFA master, I observed a strange behaviour in the GUI while running a scene of mine where I'm doing catheter navigation (similarly to BeamAdapter). Namely :
- Before, Ctrl was the default key to use interactive controls, for instance `Ctrl+uparrow` was caught by `if event['key'] == chr(19)` in my Python controller. Now `uparrow` is caught by the same line.
- Before, keeping `Ctrl+uparrow` pushed down was automatically playing the control over and over (until the key was released). Now I have to release the key (`uparrow`) and press it again to redo the control.

Talking with @fredroy, it seems that I'm not the only one to have observed this problem, so it might come from a PR undesirable side effect. Additionally, it apparently doesn't come from SofaPython3.

Regarding the master, all I can say is that the previous update I made was on the 30/09/2022 (so I didn't have this issue at the time), and that the issue appeared after my last update on the 24/11/2022. I briefly went through the merged PR in between these dates, but could find one which could be incriminated.

I'm working on Ubuntu (20.04). It is also worth mentioning that I did some other updates during the time lapse (basically, all default updates suggested by Ubuntu's Software Updater), but no major change in any software version. For this reason, I can't be sure at the moment that the issue traces back to SOFA itself.

Is anyone else experiencing this ?

**Comment by alxbilger** [comment]
If you have the time, a dichotomic strategy could help to identify (quickly) the commit responsible of this change of behavior. You could use `git bisect` to help (see https://delicious-insights.com/en/posts/git-bisect/ for example).

**Comment by Camille-K** [comment]
Thanks for the tip! It's been some time since I've used `git bisect`, so I might be mistaking, but for me the only way to judge if a commit is 'good', is to recompile and test the scene I was mentioning at each iteration of bisect, right ? In which case the process could take a bit of time...

I'll keep you updated if I go through !

**Comment by alxbilger** [comment]
Yes you're right. It can take some time, but if you really have no idea what commit is the guilty one, it is the fastest method. If you go with this process, disable the compilation of useless plugins and tests.

**Comment by chhas** [comment]
Hi, I'm working with SOFA through SofaPython3 and I've been experiencing the same change in behavior on Ubuntu 20.04 after I (re)built SOFA from the master branch on November 25th.
I didn't have this issue with a previous SOFA build from October 11th at commit ae42db86fbfc9ff70682b46dc8a2279de25361bd .
Since I also updated some Ubuntu packages and SofaPython3 in the meantime, I'm not sure what is causing this change in behavior as well.
However, `git bisect` led me to #3299, i.e. commit 18149f4407fd78359eead7b64ae23470687c40c4 from October 28th, which modified some Qt/GUI-related files like [QtGLViewer](https://github.com/sofa-framework/sofa/blob/18149f4407fd78359eead7b64ae23470687c40c4/Sofa/GUI/Qt/src/sofa/gui/qt/viewer/qgl/QtGLViewer.cpp#L826).
I'm not familiar with the code base enough to investigate further, but I also built SOFA at commit 58d37e9726187f4f8662fe4786c039443b306bd1 which is the one before #3299 and the change in interactive control behavior does not seem to be present there for me.

**Comment by Camille-K** [comment]
Thanks a lot @chhas for the investigation :raised_hands:
I'm no specialist either, but, although most changes in the commit are simply `Vector3 -> Vec3`, there are two things that seem suspicious to me, in `SofaViewer.cpp` (l.256-295) and `QtGLViewer.cpp`(l.823-860) (sorry, I can't find an easy way to link to a specific line in a commit diff view).
In both, there is a change of condition from :
```
if( isControlPressed() )
```
to
```
if (!isControlPressed() && !e->isAutoRepeat())
```
before calling a `propagateEvent` method on the `QKeyEvent` being processed.
I had a brief look: `isAutoRepeat` seems to be a Qt routine always returning an `autor` variable being set to 1 in `qevent.h`. And `isControlPressed()` refers to an internal variable `m_isControlPressed` in `SofaViewer.h`, which does what its name suggests.

At this point, it seems that these modifications could very well explain the two changes of behaviour I was describing above.
@damienmarchal, could you give us some input on this ?

**Comment by damienmarchal** [comment]
Hello @Camille-K and @chhas  thanks for very well done and helpfull analysis. 

At first sight I would say that this change has nothing to do with the vector3 to vec3 change so it must  be removed asap.
I will do that. That says the change is in fact this change was for a separated PR fixing this issue ttps://github.com/sofa-framework/sofa/issues/3384 
But as the fix was breaking it was not supposed to be integrated that way;

**Comment by alxbilger** [comment]
Can this issue be closed?

**Comment by epernod** [comment]
Fixed by PR #3562  (make sure to use the good keyword "fix "  in the PR description)
